blockchain/common_test.go

```diff
@@ -134,27 +118,25 @@ func createTestUtxoDatabase(dbName string) (*leveldb.DB, func(), error) {
 // chainSetup is used to create a new db and chain instance with the genesis
 // block already inserted.  In addition to the new chain instance, it returns
 // a teardown function the caller should invoke when done testing to clean up.
-func chainSetup(dbName string, params *chaincfg.Params) (*BlockChain, func(), error) {
+func chainSetup(t testing.TB, params *chaincfg.Params) (*BlockChain, error) {
 	if !isSupportedDbType(testDbType) {
-		return nil, nil, fmt.Errorf("unsupported db type %v", testDbType)
+		return nil, fmt.Errorf("unsupported db type %v", testDbType)
 	}
 
 	// Create a test block database.
-	db, teardownDb, err := createTestDatabase(dbName, testDbType, blockDataNet)
+	db, err := createTestDatabase(t, testDbType, blockDataNet)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	// Create a test UTXO database.
-	utxoDb, teardownUtxoDb, err := createTestUtxoDatabase(dbName + "_utxo")
+	utxoDb, teardownUtxoDb, err := createTestUtxoDatabase(t.Name() + "_utxo")
 	if err != nil {
-		teardownDb()
-		return nil, nil, err
+		return nil, err
 	}
-	teardown := func() {
+	t.Cleanup(func() {
```

Comment:
This requires Go1.18 which we can't require yet.  We always support the latest version and the previous one.  At this time, that is Go1.17 and Go1.18.

As a bit of an aside, this would also necessitate a bump of the required Go version in the module accordingly, which I don't see done in the PR, but, as mentioned, we can't make use of features that require anything later than Go 1.17 yet.

Reply:
![image](https://user-images.githubusercontent.com/20135478/158732254-9e2ca4dd-6e25-4715-8c51-97db6d303714.png)

`t.Cleanup` doesn't require Go 1.18. It was added in Go 1.14.

Reply:
Oops, you're right.  I was accidentally looking at it on the new `Fuzzer` type (https://pkg.go.dev/testing#F.Cleanup) which requires Go 1.18.  Thanks for the correction.